File: app/src/main/java/com/example/watering/investrecord/List3Adapter.java
package com.example.watering.investrecord;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import java.text.DecimalFormat;
import java.util.ArrayList;

/**
 * Created by watering on 17. 11. 17.
 */

@SuppressWarnings("DefaultFileTemplate")
class List3Adapter extends BaseAdapter {

    private final ArrayList<Info_List3> mData;
    private final LayoutInflater inflater;

    public List3Adapter(Context context, ArrayList<Info_List3> data) {
        mData = data;
        inflater = LayoutInflater.from(context);
    }

    @Override
    public int getCount() {
        return mData.size();
    }
    @Override
    public Object getItem(int position) {
        return mData.get(position);
    }
    @Override
    public long getItemId(int position) {
        return 0;
    }
    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if(convertView == null) {
            convertView = inflater.inflate(R.layout.layout_list3,parent,false);
        }

        DecimalFormat df = new DecimalFormat("#,###");
        TextView textView_1 = convertView.findViewById(R.id.textView_layout_list5_1);
        TextView textView_2 = convertView.findViewById(R.id.textView_layout_list5_2);
        TextView textView_3 = convertView.findViewById(R.id.textView_layout_list5_3);

        textView_1.setText(mData.get(position).getSpend().getDetails());
        textView_3.setText(df.format(mData.get(position).getSpend().getAmount()));

        return convertView;
    }
}

File: app/src/main/java/com/example/watering/investrecord/adapter/List3Adapter.kt
package com.example.watering.investrecord.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView

import com.example.watering.investrecord.info.InfoList3
import com.example.watering.investrecord.R

import java.text.DecimalFormat
import java.util.ArrayList

/**
 * Created by watering on 17. 11. 17.
 */

class List3Adapter(context: Context, private val mData: ArrayList<InfoList3>) : BaseAdapter() {
    private val inflater: LayoutInflater = LayoutInflater.from(context)

    override fun getCount(): Int {
        return mData.size
    }

    override fun getItem(position: Int): Any {
        return mData[position]
    }

    override fun getItemId(position: Int): Long {
        return 0
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var convertView = convertView
        if (convertView == null) {
            convertView = inflater.inflate(R.layout.layout_list3, parent, false)
        }

        val df = DecimalFormat("#,###")
        val textView_1 = convertView!!.findViewById<TextView>(R.id.textView_layout_list3_1)
        val textView_2 = convertView.findViewById<TextView>(R.id.textView_layout_list3_2)
        val textView_3 = convertView.findViewById<TextView>(R.id.textView_layout_list3_3)

        textView_1.text = mData[position].spend!!.details
        textView_3.text = df.format(mData[position].spend!!.amount.toLong())

        return convertView
    }
}
